pkg/sql/distsqlplan/aggregator_funcs.go

```diff
@@ -224,6 +224,41 @@ var DistAggregationTable = map[distsqlrun.AggregatorSpec_Func]DistAggregationInf
 			return expr.TypeCheck(nil, parser.TypeAny)
 		},
 	},
+
+	// For VARIANCE/STDDEV we need three and two intermediate values in the local and final stage,
+	// respectively:
+	//  - the local stage accumulates the SQRDIFF, SUM and the COUNT;
+	//  - the final stage calculates the FINAL_(VARIANCE|STDDEV)
+	//
+	// At a high level, this is analogous to rewriting VARIANCE(x) as SQRDIFF(x)/COUNT(x)
+	// (and STDDEV(x) as sqrt(VARIANCE(x))).
+	distsqlrun.AggregatorSpec_VARIANCE: {
+		LocalStage: []distsqlrun.AggregatorSpec_Func{
+			distsqlrun.AggregatorSpec_SQRDIFF,
+			distsqlrun.AggregatorSpec_SUM,
+			distsqlrun.AggregatorSpec_COUNT,
+		},
+		FinalStage: []FinalStageInfo{
```

Comment:
Point of interest: Instead of have a `SUM_SQDIFFS` and `SUM_INT` (for `COUNT`) stage for `VARIANCE` and `STDDEV` then tailoring a `FinalRendering` stage specific to each, I had to use a specific `FINAL_(VARIANCE|STDDEV)` aggregation stage. I ran into trailing zero issues with `STDDEV` and taking the `Pow(0.5)` of a `SUM_SQDIFFS` decimal value.